modules/extensions-xml-support/src/main/java/org/mule/runtime/extension/internal/ast/MacroExpandedComponentAst.java

```diff
@@ -107,14 +111,39 @@ public <T> Either<String, T> getValue() {
         }
 
         return (Either<String, T>) originalValue
-            .mapRight(v -> {
-              if (v instanceof String) {
-                return macroExpandedRawValue((String) v);
-              } else if (getDecorated().getModel().getName().equals(ERROR_MAPPINGS_PARAMETER_NAME)) {
-                return mapErrorMappings(getDecorated());
-              }
-              return v;
-            });
+            .mapRight(this::mapComponent);
+      }
+
+      private <T> Object mapComponent(T rawValue) {
+        if (isAnExpression(rawValue)) {
+          return macroExpandedRawValue((String) rawValue);
+        } else if (isAnErrorMappings()) {
+          return mapErrorMappings(getDecorated());
+        } else if (supportsChildDeclaration() && !isAKnownType(rawValue)) {
+          return copyMacroExpandedComponentTreeRecursively();
+        }
+        return rawValue;
+      }
+
+      private <T> boolean isAnExpression(T rawValue) {
+        // Evaluates if the parameter is not declared as a child and can be an expression so that it is resolved.
+        // In that case, the raw value is always a String.
+        return rawValue instanceof String;
+      }
+
+      private boolean isAnErrorMappings() {
+        return getDecorated().getModel().getName().equals(ERROR_MAPPINGS_PARAMETER_NAME);
+      }
+
+      private <T> boolean isAKnownType(T rawValue) {
```

Comment:
I would change the isAKnownType ->  isResolvedByMetadataVisitor.
I would add the comments to the invocation of the methods which is where the reader of the code benefits from the comment.

Reply:
I didn't change the location of the comment because I think it won't be clear where it belongs due to the formatter code and the ifs/else ifs.

Reply:
ok. I think it is clear by the rename.